app/code/ZP/LoyaltyProgram/Console/Command/Customer/AssignLoyaltyProgram.php

```diff
@@ -0,0 +1,200 @@
+
+            return 0;
+        } catch (\Exception $exception) {
+            $output->writeln($exception->getMessage());
+
+            return 1;
+        }
+    }
+
+    private function isAllCustomers(): bool
+    {
+        return $this->isAllCustomers;
+    }
+
+    /**
+     * @param array $customerIds
+     * @return array
+     * @throws \Exception
+     */
+    private function checkDataType(array $customerIds): array
+    {
+        foreach ($customerIds as $key => $customerId) {
+            if (!$this->dataValidator->isDataInteger($customerId)) {
+                $this->messageManager->setResultValue(MessageManager::WRONG_DATA, $customerId);
+                unset($customerIds[$key]);
+            }
+        }
+
+        return $this->prepareData->makeArrayKeysLikeValues($this->prepareData->arrayValuesToInteger($customerIds));
+    }
+
+    /**
+     * @param array $customers
+     * @param array $customerIds
+     * @return bool
+     * @throws \Exception
+     */
+    private function compareCustomersCount(
+        array $customers,
+        array $customerIds
+    ): bool {
+
+        if (!$customers || count($customers) !== count($customerIds)) {
+            foreach ($customerIds as $customerId) {
+                if (!array_key_exists($customerId, $customers)) {
+                    $this->messageManager->setResultValue(MessageManager::NOT_EXIST, $customerId);
+                }
+            }
+        }
+
+        return (bool)$customers;
+    }
+
+    /**
+     * @param CustomerInterface[] $customers
+     * @throws \Exception
+     */
+    private function assignLoyaltyProgramsToCustomers(array $customers): void
+    {
+        /** @var CustomerInterface $customer */
+        foreach ($customers as $customerId => $customer) {
+            $this->loyaltyProgramManagement->assignLoyaltyProgram($customer);
+            $result = $this->loyaltyProgramManagement->returnResult();
+            $this->messageManager->setResultValue($this->loyaltyProgramManagement->returnResult(), $customerId);
+        }
+    }
+
+    private function getCustomerIds(InputInterface $input)
+    {
+        $customerIds = $input->getArgument(self::CUSTOMER_IDS);
+        $this->setIsAllCustomersProperty(!$customerIds);
+
+        return $customerIds;
+    }
+
+    /**
+     * @param array $customerIds
+     * @return array
+     * @throws \Exception
+     */
+    private function validateCustomerIds(array $customerIds): array
+    {
+        return $this->isAllCustomers() ? $customerIds : $this->checkDataType($customerIds);
+    }
+
+    /**
+     * @param array $customerIds
+     * @return array
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    private function getCustomers(array $customerIds): array
+    {
+        $customers = [];
+        if (count($customerIds) === 1) {
+            $data[] = $this->customerRepository->getById($customerIds[array_key_first($customerIds)]);
+        } else {
```

Comment:
Yes, because if in console command requester didn't add any customer ID, it means that he wanted assign(reassign) if possible to ALL customers, so I don't need to add Filter to searchCriteriaBuilder. But if requester added any IDs, I will add Filter with this IDs.

Reply:
Okay, i understand you need to filter it, but read carefully my comment one more time. Do you really need to write it this way? why not to use elseif condition?

Reply:
please use elseif

Reply:
Impossible to use elseif, because :
if (count($customerIds) === 1) {
      $customers[] = // receive customer with get(); method from repository.
} else {
      if (!$this->isAllCustomers()) {
             $this->searchCriteriaBuilder->addFilter // here we add filter of customer ids to searchCriteriaBuilder!!!
      }
      
      $customers = $this->customerRepository->getList($this->searchCriteriaBuilder->create())->getItems();
      // and here we receive customers with filter or without filter in searchCriteriaBuilder according to command request!!!
}


In your version it will be like this :
if (count($customerIds) === 1) {
      $customers[] = // receive customer with get(); method from repository.
} elseif (!$this->isAllCustomers()) {
      $this->searchCriteriaBuilder->addFilter // here we add filter of customer ids to searchCriteriaBuilder!!!      
      $customers = $this->customerRepository->getList($this->searchCriteriaBuilder->create())->getItems();
      // and here we receive customers with filter or without filter in searchCriteriaBuilder according to command 
         request!!!
} 
BUT IN CASE OF $this->isAllCustomers() === true , we will not go inside elseif condition and because of this we will not get list of all customers!